include/lo2s/line_info.hpp

```diff
@@ -32,33 +32,62 @@ namespace lo2s
 {
 struct LineInfo
 {
+private:
+    static const std::string unknown_file_str;
+    static const std::string unknown_binary_str;
+    static const std::string unmapped_function_str;
+    static const std::string unknown_function_str;
+
+    static constexpr unsigned int UNKNOWN_LINE = 0;
+
     // Note: If line is not known, we write 1 anyway so the rest is shown in vampir
-    LineInfo(const std::string& fi, const std::string& fu, unsigned int l, const std::string& d)
-    : file(fi), function(fu), line(l ? l : 1), dso(boost::filesystem::path(d).filename().string())
+    // phijor 2018-11-08: I think this workaround is not needed anymore? vampir
+    // shows source code locations with line == 0 just fine.
+    LineInfo(const std::string& file, const std::string& function, unsigned int line,
+             const std::string& dso)
+    : file(file), function(function), line((line != UNKNOWN_LINE) ? line : 1), dso(dso)
+    {
+    }
+
+public:
+    LineInfo() : LineInfo(unknown_file_str, unknown_function_str, UNKNOWN_LINE, unknown_binary_str)
```

Comment:
If there are "factory" functions for everything, can't we simply delete the default constructor?

Reply:
Sure, I think we don't need that anymore.